libraries/src/CMS/Router/ApiRouter.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Joomla! Content Management System
+ *
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\CMS\ApiRouter;
+use Joomla\Router\Router;
+
+/**
+ * Joomla! API Router class
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class ApiRouter extends Router
+{
+	/**
+	 * Router instances container.
+	 *
+	 * @var    array
+	 * @since  DEPLOY_VERSION
+	 */
+	protected static $instances = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @param   array  $maps  An optional array of route maps
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct(array $maps = [])
+	{
+		parent::__construct($maps);
+	}
+
+	/**
+	 * Returns an ApiRouter object, only creating it if it
+	 * doesn't already exist.
+	 *
+	 * @param   string  $name     The name (optional) of the ApiRouter class to instantiate.
+	 * @param   array   $options  An associative array of options
+	 *
+	 * @return  ApiRouter  An ApiRouter object.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  \RuntimeException
+	 */
+	public static function getInstance($name, $options = array())
+	{
+		if (empty(self::$instances[$name]))
+		{
+			// Create a Router object
+			$classname = 'ApiRouter' . ucfirst($name);
```

Comment:
This logic seems a bit weird. This is how things used to work when we had named children (i.e. `JRouterSite`, but we only have one route for webservices). The logic needs adjusting for namespaced code anyhow

Rather than fixing that though - rather than having a `getInstance` method I think we should be provisioning this as a container provider (see examples in https://github.com/joomla/joomla-cms/tree/4.0-dev/libraries/src/CMS/Service/Provider).